ui/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container-component.test.js

```diff
@@ -87,12 +95,76 @@ describe('GasModalPageContainer Component', () => {
     jest.clearAllMocks();
   });
 
-  describe('componentDidMount', () => {
-    it('should call getGasFeeEstimatesAndStartPolling', () => {
-      jest.clearAllMocks();
-      expect(getGasFeeEstimatesAndStartPolling).not.toHaveBeenCalled();
-      wrapper.instance().componentDidMount();
-      expect(getGasFeeEstimatesAndStartPolling).toHaveBeenCalled();
+  describe('when the component mounts', () => {
+    describe('when the network does not support EIP 1559', () => {
+      it('should call updateWithAndStartPollingFor to fetch gas fee estimates', () => {
+        expect(updateWithAndStartPollingFor).not.toHaveBeenCalled();
+        const store = configureMockStore()({
+          metamask: {
+            keyrings: [],
+            identities: {},
+            networkDetails: { EIPS: {} },
+          },
+        });
+        renderWithProvider(
+          <GasModalPageContainer
+            cancelAndClose={propsMethodSpies.cancelAndClose}
+            onSubmit={propsMethodSpies.onSubmit}
+            updateCustomGasPrice={() => 'mockupdateCustomGasPrice'}
+            updateCustomGasLimit={() => 'mockupdateCustomGasLimit'}
+            gasPriceButtonGroupProps={mockGasPriceButtonGroupProps}
+            infoRowProps={mockInfoRowProps}
+            customGasPriceInHex="mockCustomGasPriceInHex"
+            customGasLimitInHex="mockCustomGasLimitInHex"
+            insufficientBalance={false}
+            disableSave={false}
+            customPriceIsExcessive={false}
+          />,
+          store,
+        );
+        expect(updateWithAndStartPollingFor).toHaveBeenCalledWith(
+          'gasFeeEstimates',
+        );
+      });
+    });
+
+    describe('when the network supports EIP 1559', () => {
+      it('should call updateWithAndStartPollingFor twice, once to get gas fee estimates and another time to gauge network congestion', () => {
+        expect(updateWithAndStartPollingFor).not.toHaveBeenCalled();
```

Comment:
I have a doubt here, since it keeps polling already why is a second call needed ?

Reply:
This was the way that `getGasFeeEstimatesAndStartPolling` worked before, so I copied the behavior over, but my guess as to why it works this way is that the timer is set to 15 seconds by default, so if we simply started polling then when the user opened the transaction confirmation screen, they wouldn't see any estimates until 15 seconds had passed. So we make the request once at the beginning manually to populate an initial set of estimates, so that the confirmation screen shows correctly, then start the timer.